setup.py

```diff
@@ -12,22 +12,89 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-import setuptools
 from setuptools import setup
+from setuptools.command.build_ext import build_ext
+from setuptools.extension import Extension
+
 import versioneer
 
 # read the contents of your README file
 from pathlib import Path
 this_directory = Path(__file__).parent
 long_description = (this_directory / "README.md").read_text()
 
+ZITI_SDK_BASE = 'https://github.com/openziti/ziti-sdk-c/releases/download'
+
+
+class GetZitilib(build_ext):
+
+    def build_extension(self, ext) -> None:
+        import platform
+        ziti_ver = self.get_sdk_version()
+        osname, arch, libname = self.get_platform()
+        sdk_distro = self.download_sdk(ziti_ver, osname, arch)
+        self.extract_zitilib(sdk_distro, libname, self.build_lib)
+
+    def get_platform(self):
+        import platform
+        osname = platform.system()
+        mach = platform.machine()
+        arch, _ = platform.architecture()
+        if osname == 'Linux':
+            if mach.startswith('arm'):
+                if arch == '32bit':
+                    mach = 'arm'
+                elif arch == '64bit':
+                    mach = 'arm64'
+            return osname, mach, 'libziti.so'
+        elif osname == 'Darwin':
```

Comment:
`elif` after `return` is not a good patter. Just do `if...return...`, then `if...return` again